### PR TITLE
bin/cake bake migration AlterSomeFieldOnSomeTable some_field:options

### PR DESCRIPTION
**Pull request by eather009**
This new Command will help to alter only one or multiple `Fields` when necessary.
Currently, we need to `Drop` a field and add the field again with new changes.
To make that easy, this new command will help to `Alter` specific fields.

Migration names can follow any of the following patterns:

> * (``/^(Alter).*(?:From)(.*)/``) Alters fields from the specified table.

**Alter a column from a table**
-----------------------------------
In the same way, you can generate a migration to alter a column by using the
command line, if the migration name is of the form "AlterXXXFromYYY":

**code-block**

    `bin/cake bake migration AlterPriceFromProducts name:float`

**Will generate**
```
use Migrations\AbstractMigration;

    class AlterPriceFromProducts extends AbstractMigration
    {
        public function change()
        {
            $table = $this->table('products');
            $table->changeColumn('name', 'float');
            $table->update();
        }
    }
```

**Documents are also updated**

### CODE REVIEW

**Comment by eather009** [comment]
@markstory , Hi, I am not sure how to resolve codecov checking.

File: docs/en/index.rst [hunk 1]
```diff
@@ -193,6 +193,7 @@ Migration names can follow any of the following patterns:
   specified table
 * (``/^(Alter)(.*)/``) Alters the specified table. An alias
   for CreateTable and AddField.
+* (``/^(Alter).*(?:From)(.*)/``) Alters fields from the specified table.
```
Comment (markstory) [inline docs/en/index.rst]: Instead of `From` should this be `On`?

Comment (eather009) [reply docs/en/index.rst]: Thank you very much. I will update and commit shortly.

**Comment by saeideng** [comment]
do you want to add more tests ?
https://github.com/cakephp/migrations/blob/master/tests/TestCase/Shell/Task/MigrationTaskTest.php

**Comment by eather009** [comment]
Oh, I should have added one test case here. I will add that soon and commit again shortly

File: src/View/Helper/MigrationHelper.php [hunk 2]
```diff
@@ -103,6 +103,10 @@ public function columnMethod($action)
             return 'removeColumn';
         }
 
+        if ($action === 'alter_field') {
+            return 'changeColumn';
+        }
+
```
Comment (saeideng) [inline src/View/Helper/MigrationHelper.php]: just a question : should not apply this to `indexMethod()` too ?

Comment (eather009) [reply src/View/Helper/MigrationHelper.php]: Thank you for your quick response. I didn't check about indexMethod. Logically user can create index through alter. So, it should. I will check and response shortly. 
Thank you very much

Comment (eather009) [reply src/View/Helper/MigrationHelper.php]: Just checked the functionality of `indexMethod`. Here have two observations. 
Someone can set index while creating and later just change the field type or any other attributes, keeping the indexed. 

Someone can create the index again. So, I think, for modifying index, should use another function to drop index. What do you say? This is just my thoughts.

**Comment by eather009** [comment]
Mr, @saeideng is there anything you suggest?

File: docs/en/index.rst (hunk 1 above)
Comment (eather009) [reply docs/en/index.rst]: Mr, @markstory , I have updated and pushed new updates already. 

Hope to see you at CakeFest in Tokyo. :) I will attend in Conference.

**Comment by saeideng** [comment]
@eather009  I did not test it yet

**Comment by eather009** [comment]
Please let me know if you find anything. Thank you very much.

**Pull request merged**